# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1533382529 -7200
#      Sat Aug 04 13:35:29 2018 +0200
# Node ID a3f2b06c599ae3e8759ccdd65be939a9347a141c
# Parent  b1e0e58971f90ac92db7e5fe33cdd91c4734341d
fix broken file closing when deleted/renamed in file browser

* file-editor-tab.cc (file_has_changed): only test last modified time when
  the file should not be closed

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -2312,17 +2312,17 @@ namespace octave
     // User canceled, allow editing again.
     _edit_area->setReadOnly (false);
   }
 
   void file_editor_tab::file_has_changed (const QString&, bool do_close)
   {
     bool file_exists = QFile::exists (_file_name);
 
-    if (file_exists)
+    if (file_exists && ! do_close)
       {
         // Test if file is really modified or if just the timezone has
         // changed. In the latter, just return without doing anything
         QDateTime modified = QFileInfo (_file_name).lastModified ().toUTC ();
 
         if (modified <= m_last_modified)
           return;
 
